static/absolute/js/searchPage.js

```diff
@@ -147,57 +157,211 @@ function showRestaurants(restaurantsResponse) {
     return;
   }
 
-  //This will be used to create a new div for every restaurant returned by the Places Library:
-  allRestaurants.results.forEach((restaurant) => {
-    let restaurantDiv = document.createElement("div");
-    restaurantDiv.classList.add("restaurant-card");
+  //Used to get additional information about the restaurant results.
+  const fields = "url,formatted_phone_number,website,review";
+
+  //Create a new div for every restaurant returned by the Places Library:
+  let restaurantIndex = 1;
+  for (restaurant of allRestaurants.results) {
+    //Get additional details for every restaurant using that restaurant's place id.
+    let placeDetailsResponse = await (
+      await fetch(
```

Comment:
2 notes:

1. This will wait for each place details response to finish sequentially, which will be quite slow. It would be nice to parallelize the place details requests. However, since we're so close to the deadline, we might not have the time.

2. Would it make sense to do the place details requests on the server and pass the information back to the client? It would probably be easier to parallelize on the server, since the ordering of the restaurants that aren't returned doesn't matter when you're on the server, but, on the client we need to know which data corresponds to which HTML element.